File: c/hash-table/island-perimeter.c
int islandPerimeter(int** grid, int gridRowSize, int gridColSize) {
    int perimeter = 0;
    int i,j;

    for(i = 0; i < gridRowSize; ++i){
        for(j = 0; j < gridColSize; ++j){
            if(grid[i][j] == 0)
                continue;

            if(j == 0 || grid[i][j-1] == 0)
                perimeter++;
            if(i == 0 || grid[i-1][j] == 0)
                perimeter++;
            if(j == gridColSize -1 || grid[i][j+1] == 0)
                perimeter++;
            if(i == gridRowSize -1 || grid[i+1][j] == 0)
                perimeter++;
        }
    }

    return perimeter;
}

File: python/hash-table/island-perimeter.py
class Solution(object):
    def islandPerimeter(self, grid):
        row = len(grid)
        col = len(grid[0]) if row else 0
        p = 0
        for i in range(row):
            for j in range(col):
                if grid[i][j] == 0:
                    continue
                p += 4

                if i > 0 and grid[i-1][j]:
                    p -= 2
                if j > 0 and grid[i][j-1]:
                    p -= 2

        return p
